### PR TITLE
Feature/approve announcement

### PR DESCRIPTION
**Pull request by dboling19**
# Description

Adds the ability for moderators and above to approve announcements. New announcements will be created with approval set to 0 as false. Moderators and admins can access the new page at `~/moderation/announcements ` where a simple button-press will set an announcement to approved, which will only then be displayed on the overview page.

## Fixes 

Fixed old tests, mainly the ones for recurrence, to work better with the new queries and reformatted it for readability and processing efficiency.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

I created a new test `test_approval` which queries for `*today_approved()` and saves the count. Then it creates an un-approved announcement, checks that there is no increase in the count, and creates an approved announcement, then checks that the count increased by 1.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/Controller/OverviewController.php
Comment (davidwkdavidwk) [inline src/Controller/OverviewController.php:281]: Your function is named `approve_announcement()` but this block un-approves the announcement.  Could you either remove this block so that if you call the function on an already approved announcement it doesn't remove the approval OR update the function name and description to reflect the fact that the function toggles the approval status rather than just approving it?  Does that make sense?

File: src/Repository/AnnouncementRepository.php
Comment (davidwkdavidwk) [inline src/Repository/AnnouncementRepository.php:29]: Would it work to add an argument to this function definition like `$approved=true` and then use an `if` statement in the function to selectively add the `->andWhere('a.approval = 1')` condition?  You could do the same thing for the `find_today()` function which would remove the need for a separate function named `find_today_approved()`.  Does that make sense?

**Review by davidwkdavidwk — CHANGES_REQUESTED** [review]

File: src/Controller/OverviewController.php
Comment (dboling19) [reply src/Controller/OverviewController.php:281]: Function name and description modified to better fit the purpose of the function, which was intended to serve as both approval and denial of announcements, assuming it was approved accidentally or needs removal.

File: src/Repository/AnnouncementRepository.php
Comment (dboling19) [reply src/Repository/AnnouncementRepository.php:29]: Can I put a condition directly into the query line there, or would I need to make "two separate queries" in one function, just selected by a condition?

File: src/Repository/AnnouncementRepository.php
Comment (davidwkdavidwk) [inline src/Repository/AnnouncementRepository.php:95]: I don't think this function is needed anymore since `find_today()` can handle both approved and unapproved announcements, right?

File: src/Controller/OverviewController.php
Comment (davidwkdavidwk) [inline src/Controller/OverviewController.php:97]: You should be able to go back to `find_today()` since that defaults to only finding approved announcements.

File: src/Controller/OverviewController.php
Comment (davidwkdavidwk) [inline src/Controller/OverviewController.php:246]: You'll probably need to change this to `find_today($approval=0)` or it won't find any un-approved announcements.  This is the kind of change that a test should catch.  Do you have tests checking whether they can find un-approved announcements?

File: tests/Repository/AnnouncementRepositoryTest.php
Comment (davidwkdavidwk) [inline tests/Repository/AnnouncementRepositoryTest.php:369]: This should also change to `find_today()`

File: tests/Repository/AnnouncementRepositoryTest.php
Comment (davidwkdavidwk) [inline tests/Repository/AnnouncementRepositoryTest.php:392]: And this should change to `find_today()`

File: tests/Repository/AnnouncementRepositoryTest.php
Comment (davidwkdavidwk) [inline tests/Repository/AnnouncementRepositoryTest.php:373]: Could you add another test in here where you do a `find_today($approved=0)` and then assert that the count should be one more than the pre count?

**Review by davidwkdavidwk — CHANGES_REQUESTED** [review]
There are some additional changes to finish up the `find_today()` changes you made.  Thanks

File: src/Controller/OverviewController.php
Comment (dboling19) [reply src/Controller/OverviewController.php]: I don't have any controller tests yet. I only have tests dedicated to checking that the database and query functions work properly.

**Review by davidwkdavidwk — APPROVED** [review]
👍

**Pull request merged**
